src/ir_York.cpp

```diff
@@ -0,0 +1,362 @@
+void IRYorkAc::stateReset() {
+  // This resets to a known-good state.
+  setRaw(kYorkKnownGoodState);
+}
+
+/// Set up hardware to be able to send a message.
+void IRYorkAc::begin(void) { _irsend.begin(); }
+
+/// Get the raw state of the object, suitable to be sent with the appropriate
+/// IRsend object method.
+/// @return A copy of the internal state.
+uint8_t *IRYorkAc::getRaw(void) {
+  calcChecksum();
+  return _.raw;
+}
+
+/// Set the internal state from a valid code for this protocol.
+/// @param[in] new_code A valid code for this protocol.
+/// @param[in] length Length of the code in bytes.
+void IRYorkAc::setRaw(const uint8_t new_code[], const uint16_t length) {
+  std::memcpy(_.raw, new_code, length);
+}
+
+#if SEND_YORK
+/// Send the current internal state as an IR message.
+/// @param[in] repeat Nr. of times the message will be repeated.
+void IRYorkAc::send(const uint16_t repeat) {
+  _irsend.sendYork(getRaw(), kYorkStateLength, repeat);
+}
+#endif  // SEND_YORK
+
+/// Get the current operation mode setting.
+/// @return The current operation mode.
+uint8_t IRYorkAc::getMode(void) const {
+  return _.Mode;
+}
+
+/// Set the desired operation mode.
+/// @param[in] mode The desired operation mode.
+void IRYorkAc::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kYorkFan:
+    case kYorkCool:
+    case kYorkHeat:
+    case kYorkDry:
+      _.Mode = mode;
+      break;
+    default:
+      _.Mode = kYorkAuto;
+  }
+  setFan(getFan());  // Ensure the fan is at the correct speed for the new mode.
+}
+
+/// Convert a stdAc::opmode_t enum into its native mode.
+/// @param[in] mode The enum to be converted.
+/// @return The native equivalent of the enum.
+uint8_t IRYorkAc::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kCool: return kYorkCool;
+    case stdAc::opmode_t::kHeat: return kYorkHeat;
+    case stdAc::opmode_t::kDry:  return kYorkDry;
+    case stdAc::opmode_t::kFan:  return kYorkFan;
+    default:                     return kYorkAuto;
+  }
+}
+
+/// Convert a native mode into its stdAc equivalent.
+/// @param[in] mode The native setting to be converted.
+/// @return The stdAc equivalent of the native setting.
+stdAc::opmode_t IRYorkAc::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kYorkCool: return stdAc::opmode_t::kCool;
+    case kYorkHeat: return stdAc::opmode_t::kHeat;
+    case kYorkDry:  return stdAc::opmode_t::kDry;
+    case kYorkFan:  return stdAc::opmode_t::kFan;
+    default:        return stdAc::opmode_t::kAuto;
+  }
+}
+
+/// Set the speed of the fan.
+/// @param[in] speed The desired setting.
+/// @note The fan speed is locked to Low when in Dry mode, to auto when in auto
+/// mode. "Fan" mode has no support for "auto" speed.
+void IRYorkAc::setFan(const uint8_t speed) {
+  switch (getMode()) {
+    case kYorkDry:
+      _.Fan = kYorkFanLow;
+      break;
+    case kYorkFan:
+      _.Fan = std::min(speed, kYorkFanHigh);
+      break;
```

Comment:
You can remove this clause, it is redundant.

Reply:
this cannot be removed. In "Fan" mode the "Auto" speed is not available, hence it is capped to "High" in this case.

Reply:
Yep. I missed that.